include/zephyr/net/http/server.h

```diff
@@ -162,12 +170,28 @@ enum http_data_status {
 	HTTP_SERVER_DATA_FINAL = 1,
 };
 
+/** @brief Status of captured request headers */
+enum http_header_status {
+	HTTP_HEADER_STATUS_OK,      /**< All available headers were successfully captured. */
+	HTTP_HEADER_STATUS_DROPPED, /**< One or more headers were dropped due to lack of space. */
+	HTTP_HEADER_STATUS_NONE,    /**< No header status is available. */
+};
+
 /** @brief HTTP header representation */
 struct http_header {
 	const char *name;  /**< Pointer to header name NULL-terminated string. */
 	const char *value; /**< Pointer to header value NULL-terminated string. */
 };
 
+/** @brief HTTP request context */
+struct http_request_ctx {
+	uint8_t *data;                          /** HTTP request data */
+	size_t data_len;                        /** Length of HTTP request data */
+	struct http_header *headers;            /** Array of HTTP request headers */
+	size_t header_count;                    /** Array length of HTTP request headers */
+	enum http_header_status headers_status; /** Status of HTTP request headers */
```

Comment:
Not sure about doxygen comments, but do we need to set them `/**< ..... */` here (as they are in the same line as what is being documented)?

Reply:
Good point, I just copied the comment formatting from the `struct http_response_ctx` but it seems this is also wrong, and has comments associated with the wrong fields in the online docs.

Will update to fix both of these.

Reply:
it's correct this way :) it's the syntax for when the comment is put *after* the element it applies to, i.e. useful when one wants to make everything fit on a single line
https://www.doxygen.nl/manual/docblocks.html#memberdoc

Reply:
oops yeah sorry, looks like I'm blind today. @jukkar's suggestion IS the correct way :)